pkg/apis/openebs.io/v1alpha1/versionDetails.go

```diff
@@ -30,7 +30,7 @@ var (
 		"1.8.0": true, "1.9.0": true, "1.10.0": true, "1.11.0": true,
 		"1.12.0": true,
 	}
-	validDesiredVersion = version.GetVersion()
+	validDesiredVersion = strings.Split(version.GetVersion(), "-")[0]
```

Comment:
Q: When we perform an upgrade of the OpenEBS version will still deleting old version runtask and CASTemplates?

Reply:
Yes on upgrade the old runtasks and castemplates are cleaned.